### ISSUE TITLE
How do I install from url using choco_installed_packages in default.config.yml?

### ISSUE THREAD

**Issue by Simbaclaws** [AlexNabokikh/windows-playbook#44]
Hi there,

I'm trying to setup my gaming pc using your ansible playbook.
And I'm looking to install some software that isn't in the chocolatey package repository.

Do you perhaps know how I can use this ansible playbook to download a msi from the internet and install it? I'm a bit new to ansible, sorry for the issue creation.

Sorry if this is not meant to be put in this issue list.

**Comment by Simbaclaws** [comment]
Hi Alex,

I forked your repo for my personal setup of my gaming pc.
Does the ansible.windows.win_package only work for windows packages from microsoft or can I use a dynamic url from some other source on the web? And is it required to specify a product_id?

I won't need any of this urgently, and if I did, I'd use it on my own forked repo because I'd have my own personal setup...
Thank you so much for making this playbook, it's probably going to help me save some time.

**Comment by AlexNabokikh** [comment]
@Simbaclaws `ansible.windows.win_package` can be used with any supported packages (.exe, .msi, .msp, .appx, .appxbundle, .msix, and .msixbundle.) from any vendor. You can find full module documentation [here](https://docs.ansible.com/ansible/latest/collections/ansible/windows/win_package_module.html).
Out of curiosity what software installation do you wan to automate?

**Comment by Simbaclaws** [comment]
Mostly my current setup for a develper + gaming machine, I've already created a huge list of software to install with chocolatey.

You can see my adjusted default.config.yml in my fork of your repo. I already pushed it.

**Comment by Simbaclaws** [comment]
Sorry but I have another question, is there a way to run the playbook on my local windows machine by fetching it from the repository letting it provision itself somehow? Instead of having a workstation that provisions it through an ip address?

**Comment by AlexNabokikh** [comment]
@Simbaclaws, you can do this from within the WSL2 Linux machine. Here is a [tutorial](https://www.thomaspreischl.de/ansible-wsl-windows/) that you can use to configure your windows machine. 
Unfortunately, you cannot install ansible directly to a windows machine as you usually do on Linux or MacOS.

**Comment by Simbaclaws** [comment]
So there is no ansible-pull command I could use for within powershell for example? I'm looking to create a ansible-pull command that basically pulls a ansible playbook and configures the local machine by inserting a single command... I'll do some research to see if this is at all available in windows somehow, thank you for your time.

**Comment by Simbaclaws** [comment]
I just automated the installation of the windows features and wsl as well as runnning, cloning etc the playbook automatically so you simply have 1 single command that provisions a windows pc from itself. The script basically asks for your local admin username and password at some point, and does now all of the installation for you.

You might want to checkout my repository fork on this :)

This does mean I had to remove all the reboots from the yml's... Because it runs locally now and not from another machine.

This should work for a clean windows install... I'm testing that soon.

I also have to do some cleanup afterwards.
